From 106221a73332f51dbd40e0e62a56ab67917ab654 Mon Sep 17 00:00:00 2001
From: Marcel Admiraal <madmiraal@users.noreply.github.com>
Date: Thu, 14 Nov 2019 18:06:57 +0100
Subject: [PATCH] Update .appveyor.yml.

- Set options to be the same as travis builds,
  except use warnings=all instead of warnings=extra.
- Fail on warnings as per travis builds.
---
 .appveyor.yml | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/.appveyor.yml b/.appveyor.yml
index 05e7094981d..3f6e9320507 100644
--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -7,10 +7,19 @@ environment:
   PYTHON: C:\Python38
   SCONS_CACHE_ROOT: "%HOME%\\scons_cache"
   SCONS_CACHE_LIMIT: 1024
+  OPTIONS: "debug_symbols=no verbose=yes progress=no"
+  EXTRA_ARGS: "warnings=all werror=yes"
   matrix:
-    - GD_PLATFORM: windows
-      TOOLS: yes
-      TARGET: release_debug
+  - GD_PLATFORM: windows
+    TARGET: release_debug
+    TOOLS: yes
+# Disabled for performance reasons until master is more stable.
+#  - GD_PLATFORM: windows
+#    TARGET: release
+#    TOOLS: no
+
+matrix:
+  fast_finish: true
 
 init:
   - ps: if ($env:APPVEYOR_REPO_BRANCH -ne "master") { $env:APPVEYOR_CACHE_SKIP_SAVE = "true" }
@@ -29,4 +38,4 @@ before_build:
   - set "SCONS_CACHE=%SCONS_CACHE_ROOT%\%APPVEYOR_REPO_BRANCH%"
 
 build_script:
-  - scons platform=%GD_PLATFORM% target=%TARGET% tools=%TOOLS% debug_symbols=no verbose=yes progress=no gdnative_wrapper=yes
+  - scons platform=%GD_PLATFORM% target=%TARGET% tools=%TOOLS% %OPTIONS% %EXTRA_ARGS%

